utils/useDataFetching.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+export default function useDataFetching(client, metricId, MetricQuery) {
```

Comment:
Then we can remove those arguments and leave only the `metricId`

```suggestion
export default function useDataFetching(metricId) {
```